### PR TITLE
Support plugin type in `docker inspect`

### LINKED ISSUE #28946: [Feature request] Support plugins in `docker inspect`
#23614 added support for "all" types to `docker inspect`.
Now that the plugin API is no longer experimental, plugins could also be inspectable from `docker inspect`.
This seems natural because just like the other supportet types, there is also a `docker plugin inspect`.

I suggest to add `docker inspect --type plugin`.

Related: #28735, #28789. Maybe @yongtang is interested?

### PR DESCRIPTION
**Pull request by yongtang**
**- What I did**

This fix tries to address the proposal raised in #28946 to support plugins in `docker inspect`.

The command `docker inspect` already supports "container", "image", "node", "network", "service", "volume", "task". However, `--type plugin` is not supported yet at the moment.

**- How I did it**

This fix address this issue by adding the support of `--type plugin` for `docker inspect`.

**- How to verify it**

An additional integration test has been added to cover the changes.

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

3840x2160-animals_kitten_cute_kitten_cat-14319

This fix fixes #28946.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

**Comment by LK4D4** [comment]
LGTM

**Comment by vieux** [comment]
LGTM ping @tiborvass @anusha-ragunathan

**Review by tiborvass — CHANGES_REQUESTED** [review]
@yongtang this doesn't handle the default `:latest` tag. `docker inspect tiborvass/sample-volume-plugin` should work after installing it.

**Comment by yongtang** [comment]
Thanks @tiborvass for the review. This PR is loosely related to #28789 
In case of plugin name without tag. If #28789 is merged in then the issue is resolved.

I added a couple of test cases for plugin name without tag. The test cases will fail at the moment, but should be able to pass after #28789 is in.

Alternatively, if we want to defer the merging of #28789 after 1.13, then I could also update this PR to only include related changes to make the test cases (plugin name without tag) pass for this PR.

Please take a look at this PR and #28789.

**Comment by albers** [comment]
Very nice, but as @tiborvass mentioned, proper treatment of plugins without version (defaulting to latest) is needed. Perhaps #28789 could go into 1.13.0 as well?

File: integration-cli/docker_cli_inspect_test.go
Comment (anusha-ragunathan) [inline integration-cli/docker_cli_inspect_test.go]: Needs a `--type plugin`. Also, how's this different from the inspect command at L426?

Comment (anusha-ragunathan) [inline integration-cli/docker_cli_inspect_test.go]: Ditto for L439.

**Comment by anusha-ragunathan** [comment]
LGTM. We can merge this after 28789 is merged.

**Comment by yongtang** [comment]
@tiborvass @anusha-ragunathan Since #28789 has been merged in, I did a rebase to this PR.

Now it is possible to specify `docker inspect ` without the plugin tag. It is shown in the case of the added integration tests.

Please take a look and let me know if there are any issues.

**Comment by yongtang** [comment]
The Jenkins failure is caused by:
```
21:06:49 Set build name.
21:06:49 Workspace is not accessible
21:06:49 Build was aborted
21:06:49 [PostBuildScript] - Execution post build scripts.
21:06:49 Notifying endpoint 'HTTP:https://leeroy.dockerproject.org/notification/jenkins'
21:06:50 Finished: ABORTED
```

I think triggering the Jenkins for WindowsRS1 again will likely make it pass.

**Comment by albers** [comment]
Thanks @yongtang!

**Pull request merged**
